azext_edge/edge/providers/orchestration/clone.py

```diff
@@ -0,0 +1,1419 @@
+                },
+            },
+        }
+        if self.resource_group:
+            result["resourceGroup"] = self.resource_group
+        if self.subscription:
+            # TODO: verify
+            # result["subscription"] = self.subscription
+            pass
+        if self.parameters:
+            input_param_map = {}
+            template_param_map = {}
+
+            for param in self.parameters:
+                target_value = (
+                    self.parameters[param]["value"]
+                    if "value" in self.parameters[param]
+                    else f"[parameters('{param}')]"
+                )
+                input_param_map[param] = {"value": target_value}
+                template_param_map[param] = {"type": self.parameters[param]["type"]}
+
+            result["properties"]["parameters"] = input_param_map
+            result["properties"]["template"]["parameters"] = template_param_map
+        if self.depends_on:
+            result["dependsOn"] = list(self.depends_on)
+        return result
+
+
+class ResourceContainer:
+    def __init__(
+        self,
+        api_version: str,
+        resource_state: dict,
+        depends_on: Optional[Iterable[str]] = None,
+        config: Optional[Dict[str, Any]] = None,
+    ):
+        self.api_version = api_version
+        self.resource_state = resource_state
+        if depends_on:
+            depends_on = list(depends_on)
+        self.depends_on = depends_on
+        if not config:
+            config = {}
+        self.config = config
+
+    def _prune_resource(self):
+        filter_keys = {
+            "id",
+            "systemData",
+        }
+        self.resource_state = self._prune_resource_keys(filter_keys=filter_keys, resource=self.resource_state)
+        filter_keys = {
+            "provisioningState",
+            "currentVersion",
+            "statuses",
+            "status",
+        }
+        self.resource_state["properties"] = self._prune_resource_keys(
+            filter_keys=filter_keys, resource=self.resource_state["properties"]
+        )
+
+    def _prune_identity(self):
+        filter_keys = {"principalId"}
+        if "identity" in self.resource_state:
+            self.resource_state["identity"] = self._prune_resource_keys(
+                filter_keys=filter_keys, resource=self.resource_state["identity"]
+            )
+
+    @classmethod
+    def _prune_resource_keys(cls, filter_keys: set, resource: dict) -> dict:
+        result = {}
+        for key in resource:
+            if key not in filter_keys:
+                result[key] = resource[key]
+        return result
+
+    def _apply_cl_ref(self):
+        if "extendedLocation" in self.resource_state:
+            self.resource_state["extendedLocation"]["name"] = TEMPLATE_EXPRESSION_MAP["customLocationId"]
+
+    def _apply_nested_name(self):
+        def __extract_suffix(path: str) -> str:
```

Comment:
nit: have not seen a double _ function in the start only

Reply:
The double is meant to convey private, only accessed by the encompassing private function but I'll change it to single underscore, so the reader does not think there is an additional special meaning.